src/Exceptionless.Core/Services/StackService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Exceptionless.Core.Extensions;
+using Exceptionless.Core.Repositories;
+using Foundatio.Caching;
+using Foundatio.Utility;
+using Microsoft.Extensions.Logging;
+
+namespace Exceptionless.Core.Services {
+    public class StackService {
+        private readonly ILogger<UsageService> _logger;
+        private readonly IStackRepository _stackRepository;
+        private readonly ICacheClient _cache;
+
+        public StackService(IStackRepository stackRepository, ICacheClient cache, ILoggerFactory loggerFactory = null) {
+            _stackRepository = stackRepository;
+            _cache = cache;
+            _logger = loggerFactory.CreateLogger<UsageService>();
+        }
+
+        public async Task IncrementStackUsageAsync(string organizationId, string projectId, string stackId, DateTime minOccurrenceDateUtc, DateTime maxOccurrenceDateUtc, int count) {
+            if (String.IsNullOrEmpty(organizationId) || String.IsNullOrEmpty(projectId) || String.IsNullOrEmpty(stackId) || count == 0)
```

Comment:
Should probably throw a ArgumentNullException if any of these strings are null and bail if the count is less than or equal to 0